app/cli/internal/telemetry/telemetry.go

```diff
@@ -49,28 +50,57 @@ func NewCommandTracker(client Client) *CommandTracker {
 
 // Track sends a command event to the telemetry.Client.
 func (t *CommandTracker) Track(ctx context.Context, cmd string, tags Tags) error {
+	// Ensure a valid client is available.
 	if t.client == nil {
 		return nil
 	}
 
-	computedTags := loadDefaultTags()
-	// Add on top of the default tags the ones passed as argument.
-	// TODO: Add a way to avoid the override default tags.
-	for k, v := range tags {
-		computedTags[k] = v
-	}
+	// Load default tags and merge with user-provided tags.
+	computedTags := mergeTags(loadDefaultTags(), tags)
+
+	// Set the command tag.
 	computedTags["command"] = cmd
 
-	var id string
-	id, err := machineid.ProtectedID("chainloop")
-	if err != nil {
-		// If the machine ID is not available, use a default ID otherwise the underlying library will fail.
-		id = "default-id"
-	}
+	// Determine the user ID.
+	id := DetermineUserID(computedTags)
 
+	// Track the event with computed tags.
 	return t.client.TrackEvent(ctx, commandTrackerEventName, id, computedTags)
 }
 
+// Merges two tag maps.
+func mergeTags(defaultTags, userTags Tags) Tags {
+	result := make(Tags)
+	for k, v := range defaultTags {
+		result[k] = v
+	}
+	for k, v := range userTags {
+		result[k] = v
+	}
+	return result
+}
+
+// DetermineUserID Determines the user ID using available information or generates a random one.
+func DetermineUserID(tags Tags) string {
```

Comment:
`DetermineUserID` can be downcase if you move the tests to the same package

Reply:
I've added it into a separated packaged because if added to the same package it creates an import cycle between the mocks and the implementation:
```
FAIL	github.com/chainloop-dev/chainloop/app/cli/internal/telemetry [setup failed]
package github.com/chainloop-dev/chainloop/app/cli/internal/telemetry
	imports github.com/chainloop-dev/chainloop/app/cli/internal/telemetry/mocks
	imports github.com/chainloop-dev/chainloop/app/cli/internal/telemetry: import cycle not allowed in test
```

Reply:
You are right, this is the problem of using the mocks generated by mockery :(

A workaround is to create another file, `telemerty_unit_test.go` or similar and have these unit tests. 

In general when I end up writing code in a specific way just so it works in the tests, I realize that I might be doing smth wrong and re-evaluate. But up to you. 

Thansk!

Reply:
>A workaround is to create another file, telemerty_unit_test.go or similar and have these unit tests.

But the problem would still be the same, the method will still be public right?

I mean for me, I wouldn't even create a test for this specific function since they are covered by the tests on the `Track` method that includes it.